rulesets/google-enhanced-ecommerce-ga4.js

```diff
@@ -0,0 +1,167 @@
+  {
+    "id": "fs-ga4-e-commerce-select_item",
+    "source": "dataLayer",
+    "operators": [
+      { "name": "query", "select": "$[?(event=select_item)]" },
+      { "name": "query", "select": "$.ecommerce.items[0]" },
+      { "name": "convert", "enumerate": true },
+      { "name": "insert", "value": "select_item" }
+    ],
+    "destination": "FS.event"
+  },
+  {
+    "id": "fs-ga4-e-commerce-view_item",
+    "source": "dataLayer",
+    "operators": [
+      { "name": "query", "select": "$[?(event=view_item)]" },
+      { "name": "query", "select": "$.ecommerce.items[0]" },
+      { "name": "convert", "enumerate": true },
+      { "name": "insert", "value": "view_item" }
+    ],
+    "destination": "FS.event"
+  },
+  {
+    "id": "fs-ga4-e-commerce-add_to_cart",
+    "source": "dataLayer",
+    "operators": [
+      { "name": "query", "select": "$[?(event=add_to_cart)]" },
+      { "name": "query", "select": "$.ecommerce.items[0]" },
+      { "name": "convert", "enumerate": true },
+      { "name": "insert", "value": "add_to_cart" }
+    ],
+    "destination": "FS.event"
+  },
+  {
+    "id": "fs-ga4-e-commerce-remove_from_cart",
+    "source": "dataLayer",
+    "operators": [
+      { "name": "query", "select": "$[?(event=remove_from_cart)]" },
+      { "name": "query", "select": "$.ecommerce.items[0]" },
+      { "name": "convert", "enumerate": true },
+      {
+        "name": "insert", "value": "remove_from_cart"
+      }
+    ],
+    "destination": "FS.event"
+  },
+  {
+    "id": "fs-ga4-e-commerce-view_promotion",
+    "source": "dataLayer",
+    "operators": [
+      { "name": "query", "select": "$[?(event=view_promotion)]" },
+      { "name": "query", "select": "$.ecommerce.items[0]" },
+      { "name": "convert", "enumerate": true },
+      { "name": "insert", "value": "view_promotion" }
+    ],
+    "destination": "FS.event"
+  },
+  {
+    "id": "fs-ga4-e-commerce-select_promotion",
+    "source": "dataLayer",
+    "operators": [
+      { "name": "query", "select": "$[?(event=select_promotion)]" },
+      { "name": "query", "select": "$.ecommerce.items[0]" },
+      { "name": "convert", "enumerate": true },
+      { "name": "insert", "value": "select_promotion" }
+    ],
+    "destination": "FS.event"
+  },
+  {
+    "id": "fs-ga4-e-commerce-purchase",
+    "source": "dataLayer",
+    "operators": [
+      { "name": "query", "select": "$[?(event=purchase)]" },
+      { "name": "query", "select": "$.ecommerce" },
+      { "name": "rename", "properties": { "items": "products" } },
+      { "name": "convert", "enumerate": true },
+      { "name": "insert", "value": "Order Completed" }
+    ],
+    "destination": "FS.event"
+  },
+  {
+    "id": "fs-gtg4-e-commerce-select_item",
+    "source": "dataLayer",
+    "operators": [
+      { "name": "query", "select": "$[(0,1,2)]" },
+      { "name": "query", "select": "$[?(1=select_item)]" },
+      { "name": "query", "select": "$[2].items[0]" },
```

Comment:
Could `items` ever be empty? Is this a test case we should capture?

Reply:
I believe items can be an empty string `""`

Reply:
@PerStirpes Can you link to any supporting documentation or evidence from a live website?

Reply:
Going to resolve this as we have a test that verifies empty array behavior. [Enhanced Ecommerce specifies](https://developers.google.com/analytics/devguides/collection/ga4/reference/events?client_type=gtm#purchase) that `items` is an array and is required. It feels like we're covering the spec well and can adapt if we get feedback from real world use.